src/Components/Star/StarRating/StarRating.js

```diff
@@ -0,0 +1,197 @@
+import React, { Component } from 'react';
+import EachStar from './EachStar/EachStar';
+import './StarRating.scss';
+
+export default class StarRating extends Component {
+  constructor() {
+    super();
+    this.state = {
+      rateValue: [false, false, false, false, false],
+      hoverRateValue: [false, false, false, false, false],
+      isHover: false,
+    };
+  }
+
+  componentDidMount = () => {
+    fetch('http://localhost:1313/rating/7')
+      .then(res => res.json())
+      .then(data => {
+        const currentRateValue = Array(5)
+          .fill(false)
+          .fill(true, 0, data.rating);
+
+        this.setState({
+          rateValue: currentRateValue,
+        });
+      });
+  };
+
+  handleStarClick = clickedIndex => {
+    const prevRateValue = [...this.state.rateValue];
+    const isClickedStarActive = prevRateValue[clickedIndex];
+    const isNextStarActive = prevRateValue[clickedIndex + 1];
+
+    if (isClickedStarActive && isNextStarActive) {
+      prevRateValue.fill(false, clickedIndex + 1);
+
+      const rating = prevRateValue.filter(value => value).length;
+
+      fetch('http://localhost:1313/rating/7', {
+        method: 'PATCH',
+        body: JSON.stringify({
+          rating: rating,
+        }),
+        headers: { 'Content-type': 'application/json; charset=UTF-8' },
+      }).then(() => {
+        this.props.callApi();
+      });
+
+      this.setState({
+        isHover: false,
+        hoverRateValue: [false, false, false, false, false],
+        rateValue: prevRateValue,
+      });
+
+      return;
+    }
+
+    if (isClickedStarActive) {
+      prevRateValue.fill(false, 0, clickedIndex + 1);
+
+      const rating = prevRateValue.filter(value => value).length;
+
+      fetch('http://localhost:1313/rating/7', {
+        method: 'PATCH',
+        body: JSON.stringify({
+          rating: rating,
+        }),
+        headers: { 'Content-type': 'application/json; charset=UTF-8' },
+      }).then(() => {
+        this.props.callApi();
+      });
+
+      this.setState({
+        isHover: false,
+        hoverRateValue: [false, false, false, false, false],
+        rateValue: prevRateValue,
+      });
+
+      return;
+    }
+
+    if (!isClickedStarActive) {
+      prevRateValue.fill(true, 0, clickedIndex + 1);
+
+      const rating = prevRateValue.filter(value => value).length;
+
+      fetch('http://localhost:1313/rating/7', {
+        method: 'PATCH',
+        body: JSON.stringify({
+          rating: rating,
+        }),
+        headers: { 'Content-type': 'application/json; charset=UTF-8' },
+      }).then(() => {
+        this.props.callApi();
+      });
+
+      this.setState({
+        isHover: false,
+        hoverRateValue: [false, false, false, false, false],
+        rateValue: prevRateValue,
+      });
+
```

Comment:
- 너무 많은 부분이 동일하게 반복되고 있습니다! 달라지는 부분에 대해서만 조건을 걸어주시면 코드의 양을 더 많이 줄일 수 있을 것 같습니다!
```suggestion
    if (isClickedStarActive && isNextStarActive) {
      prevRateValue.fill(false, clickedIndex + 1);
    } else if (isClickedStarActive) {
      prevRateValue.fill(false, 0, clickedIndex + 1);    
    } else if (!isClickedStarActive) {
      prevRateValue.fill(true, 0, clickedIndex + 1);    
    }
    const rating = prevRateValue.filter(value => value).length;
    fetch('http://localhost:1313/rating/7', {
      method: 'PATCH',
      body: JSON.stringify({
        rating: rating,
      }),
      headers: { 'Content-type': 'application/json; charset=UTF-8' },
    }).then(() => {
      this.props.callApi();
    });

    this.setState({
      isHover: false,
      hoverRateValue: [false, false, false, false, false],
      rateValue: prevRateValue,
    });
```